bokeh/server/tornado.py

```diff
@@ -249,6 +280,34 @@ def __init__(self,
                      websocket_max_message_size_bytes,
                      websocket_max_message_size_bytes/1024.0**2)
 
+        if get_user and get_user_async:
+            raise ValueError("Only one of get_user or get_user_async should be supplied")
+        self._get_user = get_user
+        self._get_user_async = get_user_async
+
+        if (get_user or get_user_async) and not (login_url or get_login_url):
+            raise ValueError("When user authentication is enabled, one of login_url or get_login_url must be supplied")
+
+        if (get_user or get_user_async) and not (login_url or get_login_url):
+            raise ValueError("If a get_user function is provided, login URL must also be provided")
+        if login_url and get_login_url:
+            raise ValueError("At most one of login_url or get_login_url should be supplied")
+        if login_handler and get_login_url:
+            raise ValueError("LoginHandler cannot be used with get_login_url()")
+        if login_handler and not issubclass(login_handler, RequestHandler):
+            raise ValueError("LoginHandler must be a Tornado RequestHandler")
+        # This just catches some common cases up front, let tornado barf on any others
+        if login_handler and login_url.startswith("http") and login_url.startswith("//"):
```

Comment:
Always `False`.

Reply:
Thank you!

Reply:
Also technically a legitimate relative URL could start with `http` but I guess it's not a big deal.

Reply:
No it's a fair observation, I will tighten it up. I'm not worried about false negatives here since Tornado will catch the issue if we don't, but should try to avoid any false positives.